### PR TITLE
Build Desktop Chat settings popover

### PR DESCRIPTION
**Pull request by Nek-12**
## Summary

- adds the descriptor-backed Chat Settings contract for stateless New Chat catalogs and authoritative existing-Session reads/mutations
- adds paired Task identity projection for Session facts
- adds the Desktop `ApiService.chat` adapter and one cohesive `useChatSettings` feature boundary
- renders the Settings chip/popover, Agent chooser, Supervisor, Thinking, Fast, Questions, Auto-compaction, Session navigation facts, and Session ID copy behavior
- preserves independent mutation delivery, typed diagnostic feedback, host-owned connection availability, and stale-tolerant refresh behavior

## Verification

QA/review reported no blocking correctness, robustness, architecture, safety, security, performance, or scalability findings after the remediation rounds.

Local non-test verification completed:

- Desktop lint and typecheck
- Desktop web build
- full Desktop build
- server lint
- Go build
- diff whitespace checks

Manual browser QA was not performed in this implementation lane and remains assigned to the separate QA workflow.

## Diff size

Compared with `main`:

- production code: **+2,883 / -530** — net **+2,353**, gross **3,413**
- test, generated, and docs code: **+2,522 / -129** — net **+2,393**, gross **2,651**
- total: **+5,405 / -659** — net **+4,746**, gross **6,064**

The second category consists of tests (+2,473/-107), generated metadata query code (+23/-0), and docs/skill guidance (+26/-22).

## Caveats and follow-ups

- This is a protocol hard cutover; the protocol version is raised to 152.
- KENT-420 owns mounting the exported feature into central Chat composition and supplying navigation, Context, connection availability, and refresh generation.
- KENT-627 may extend the narrow Session-to-Task projection with Task title later; this change intentionally exposes only Task short ID.
- The merged KENT-407 contract did not include the distinct New Chat catalog and complete per-Agent baselines required by the approved Settings design. The product owner authorized adding that missing contract in KENT-416.
- No Settings cache, reconnect loop, mutation replay, client compatibility resolver, or pre-Session settings persistence was added.

## Tracking

- Product task: KENT-416 — Build Chat settings popover
- GitHub issue: none linked

### CODE REVIEW

File: apps/desktop/src/features/chat/SteppedSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+            key={step}
+            style={{ left: `${String(stepPosition(stepIndex, values.length) * 100)}%` }}
+          />
+        ))}
+        <span
+          className="pointer-events-none absolute top-1/2 h-[var(--space-3)] w-[var(--space-3)] -translate-1/2 rounded-full bg-[var(--stepped-active)] transition-[left,background-color] duration-[var(--motion-fast)] motion-reduce:transition-none"
+          style={{ left: `${String(position * 100)}%`, transitionDuration: style.transitionDuration }}
+        />
+        <input
+          className="app-region-no-drag absolute inset-0 m-0 h-full w-full cursor-pointer touch-none opacity-0 disabled:cursor-not-allowed"
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/chat/SteppedSelector.tsx:88]: No change: the Desktop Chat specification explicitly excludes screen-reader interaction architecture. The selector remains focusable and implements the required pointer and keyboard-step product behavior; an accessibility-only naming API is outside this task contract.

File: apps/desktop/src/features/chat/SessionSettings.test.tsx [hunk 2]
```diff
@@ -0,0 +1,997 @@
...
+  await user.click(control);
+  expect(mutate).toHaveBeenCalledExactlyOnceWith(target, { kind: "thinking", value: "medium" });
+});
+
+it("surfaces typed Chat operation diagnostics for an ordinary mutation failure", async () => {
+  const services = createTestServices([]);
+  vi.spyOn(services.api.chat, "getSettings").mockResolvedValue(initialRead);
+  const failure = typedSettingsFailure();
+  vi.spyOn(services.api.chat, "mutateSettings").mockRejectedValue(failure);
+  const notice = vi.spyOn(ui, "showStatusToast").mockImplementation(() => undefined);
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/chat/SessionSettings.test.tsx:284]: No change: each status-toast spy is scoped to one test and explicitly restored in that test. A failing assertion already fails the suite; a file-wide restore hook would broaden mock lifecycle beyond the approved product-boundary cases.

File: apps/desktop/src/features/chat/ChatSettingsRows.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      className="rounded-[var(--radius-m)] transition-opacity duration-[var(--motion-fast)] motion-reduce:transition-none data-[disabled=true]:opacity-45"
+      disabled={reason !== undefined}
+      reason={reason}
+    >
+      <Element
+        className="flex w-full min-w-0 items-center gap-[var(--space-3)] rounded-[var(--radius-m)] px-[var(--space-2)] py-[var(--space-2)] text-left outline-none transition-colors duration-[var(--motion-fast)] motion-reduce:transition-none data-[interactive=true]:cursor-pointer data-[interactive=true]:hover:bg-[var(--color-island-2)] focus-visible:ring-[3px] focus-visible:ring-[color-mix(in_srgb,var(--color-primary)_40%,transparent)]"
+        data-interactive={onActivate !== undefined}
+        onClick={onActivate}
+        type={Element === "button" ? "button" : undefined}
+      >
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/chat/ChatSettingsRows.tsx:27]: Addressed for the rows without nested controls: Agent rows now render as buttons and retain normal focus/keyboard activation. Supervisor and switch rows intentionally remain containers because they contain focusable radio/switch controls and must not create nested buttons.

File: shared/protoapi/chat_settings_read.go [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+		return &serverapi.ChatSettingsAgentPreparationError{
+			Agent: detail.ChatSettingsAgentPreparation.Agent,
+			Category: map[pb.AgentPreparationCategory]serverapi.ChatSettingsAgentPreparationCategory{
+				pb.AgentPreparationCategory_AGENT_PREPARATION_CATEGORY_INVALID_CONFIGURATION: serverapi.ChatSettingsAgentInvalidConfiguration,
+				pb.AgentPreparationCategory_AGENT_PREPARATION_CATEGORY_PROVIDER_UNAVAILABLE:  serverapi.ChatSettingsAgentProviderUnavailable,
+				pb.AgentPreparationCategory_AGENT_PREPARATION_CATEGORY_INTERNAL_PREPARATION:  serverapi.ChatSettingsAgentInternalPreparation,
+			}[detail.ChatSettingsAgentPreparation.Category],
+		}
+	default:
+		return errors.New(value.Code)
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply shared/protoapi/chat_settings_read.go:197]: Fixed in 7b923257f. The Go descriptor decoder now returns an error wrapping sessioncontract.ErrSessionNotFound while preserving the Session identity, with focused regression coverage.

File: apps/desktop/src/api/chatSettings.test.ts [hunk 5]
```diff
@@ -0,0 +1,856 @@
...
+                  choices: [first],
+                  initialSettings: { ...first.baseline, agentRole: "missing" },
+                },
+              },
+            },
+          },
+        }),
+      },
+    ]);
+    await expect(new ApiClient(transport).chat.getSettings(newChatTarget)).rejects.toBeInstanceOf(Error);
```
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/api/chatSettings.test.ts:338]: No change: these cases intentionally exercise generated Protobuf validation inside FakeRpcTransport before adapter projection. That boundary raises the validator error, not ContractError. ContractError assertions remain reserved for validly decoded responses that violate the Desktop domain contract.

File: apps/desktop/src/features/chat/SessionSettings.test.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nek-12) [reply apps/desktop/src/features/chat/SessionSettings.test.tsx:284]: Addressed in e22dd2344. The file now restores all spies in afterEach, so a failed assertion cannot leak the status-toast mock into later cases.
